openedx/features/qverse_features/registration/signals.py

```diff
@@ -1,20 +1,83 @@
+                                                                                    'is_active': True
+                                                                                })
+                        if created:
+                            LOGGER.info('{} user has been created.'.format(edx_user.username))
+                        else:
+                            edx_user.first_name = row.get('firstname')
+                            edx_user.last_name = row.get('surname')
+                            edx_user.email = row.get('email')
+                            edx_user.save()
+                            LOGGER.info('{} user has been updated.'.format(edx_user.username))
+
+                        full_name = '{} {}'.format(edx_user.first_name, edx_user.last_name)
+                        _, created = UserProfile.objects.update_or_create(user=edx_user,
+                                                                          defaults={'name': full_name})
+                        if created:
+                            LOGGER.info('{} edx profile has been created.'.format(edx_user.username))
+                        else:
+                            LOGGER.info('{} edx profile has been updated.'.format(edx_user.username))
+
+                        department = Department.objects.get(number=row.get('departmentid'))
+                        _, created = QVerseUserProfile.objects.update_or_create(user=edx_user,
+                                                                                defaults={
+                                                                                    'current_level': row.get('levelid'),
+                                                                                    'other_name': row.get('othername'),
+                                                                                    'mobile_number': row.get('mobile'),
+                                                                                    'programme': row.get('programmeid'),
+                                                                                    'department': department,
+                                                                                    'registration_number':
+                                                                                        row.get('regno')
+                                                                                })
+                        if created:
+                            LOGGER.info('{} qverse profile has been created.'.format(edx_user.username))
+                        else:
+                            LOGGER.info('{} qverse profile has been updated.'.format(edx_user.username))
+                except (IntegrityError, ValueError):
+                    LOGGER.exception('Error while creating/updating {} user and its profiles.'.format(row.get('regno')))
+                except Department.DoesNotExist:
```

Comment:
Can we move these exception handlings to the statements where theese could be thrown? to make the `try...except` blocks smaller?

Reply:
done where it seems good and readable but not all of them.